myia/parser.py

```diff
@@ -0,0 +1,958 @@
+            with debug_inherit(name=arg.arg, location=self.make_location(arg)):
+                param_node = Parameter(function_block.graph, name=arg.arg)
+
+            if arg.annotation:
+                param_node.add_annotation(self._eval_ast_node(arg.annotation))
+
+            function_block.graph.parameters.append(param_node)
+            function_block.write(arg.arg, param_node)
+            if dflt:
+                # TODO If there are no parents (the initial function),
+                # then evaluate in the global env
+                if block is None:
+                    raise MyiaSyntaxError(
+                        "default value on the entry function",
+                        self.make_location(arg),
+                    )
+                # XXX: This might not work correctly with our framework,
+                # but we need to evaluate the default arguments in the parent
+                # context for proper name resolution.
+                dflt_node = self.process_node(block, dflt)
+                defaults_name.append(arg.arg)
+                defaults_list.append(dflt_node)
+
+        if args.vararg:
+            arg = args.vararg
+            with debug_inherit(name=arg.arg, location=self.make_location(arg)):
+                vararg_node = Parameter(function_block.graph, name=arg.arg)
+            function_block.graph.parameters.append(vararg_node)
+            function_block.write(arg.arg, vararg_node)
+        else:
+            vararg_node = None
+
+        if args.kwarg:
+            arg = args.kwarg
+            with debug_inherit(name=arg.arg, location=self.make_location(arg)):
+                kwarg_node = Parameter(function_block.graph, name=arg.arg)
+            function_block.graph.parameters.append(kwarg_node)
+            function_block.write(arg.arg, kwarg_node)
+        else:
+            kwarg_node = None
+
+        function_block.graph.varargs = vararg_node
+        function_block.graph.kwargs = kwarg_node
+        function_block.graph.defaults = dict(zip(defaults_name, defaults_list))
+        function_block.graph.kwonly = len(args.kwonlyargs)
+
+    def _create_function(self, block, node):
+        with debug_inherit(name=node.name, location=self.make_location(node)):
+            function = Function(
+                parent=block,
+                flags=self.recflags,
+            )
+        function_block = function.initial_block
+
+        self._process_args(block, function_block, node.args)
+
+        after_block = self.process_statements(function_block, node.body)
+
+        if after_block.graph.return_ is None:
+            after_block.returns(Constant(None))
+
+        if node.returns:
+            function_block.graph.return_.add_annotation(
+                self._eval_ast_node(node.returns)
+            )
+
+        return function_block
+
+    def make_condition_blocks(self, block, tn, fn):
+        with debug_inherit(name="if_true", location=self.make_location(tn)):
```

Comment:
I suppose we can leave it for another PR because it'd require some adjustments in the tests, but it'd be better to use `about(block.graph, relation="if_true", location=...)` because it would let us generate labels like `f.if_true` instead of just `if_true`, which isn't as informative.

Reply:
Yes

Reply:
Does "yes" mean you're doing it, or that we'll leave it for another PR?

Reply:
I can do it in this PR since it's not a big change.